src/Razor/test/Microsoft.VisualStudio.LanguageServices.Razor.Test/Cohost/Shared/CohostDevToolsEndpointTest.cs

```diff
@@ -0,0 +1,141 @@
+                }
+                <div>@message</div>
+                """;
+
+        await VerifyGeneratedDocumentContentsAsync(input, GeneratedDocumentKind.Formatting);
+    }
+
+    [Fact]
+    public async Task GetTagHelpers_ReturnsJson()
+    {
+        var input = """
+                @addTagHelper *, Microsoft.AspNetCore.Mvc.TagHelpers
+                <div>Test content</div>
+                """;
+
+        var razorDocument = await CreateDocumentAsync(input);
+        var endpoint = new CohostTagHelpersEndpoint(IncompatibleProjectService, RemoteServiceInvoker);
+        
+        var request = new TextDocumentIdentifier
+        {
+            DocumentUri = razorDocument.CreateDocumentUri()
+        };
+
+        var result = await endpoint.GetTestAccessor().HandleRequestAsync(request, razorDocument, DisposalToken);
+        
+        Assert.NotNull(result);
+        Assert.NotEmpty(result);
+        
+        // Verify it's valid JSON
+        var tagHelpers = JsonSerializer.Deserialize<object>(result);
+        Assert.NotNull(tagHelpers);
+    }
+
+    [Fact]
+    public async Task GetSyntaxTree_ReturnsTree()
+    {
+        var input = """
+                @{
+                    var message = "Hello World";
+                }
+                <div>@message</div>
+                """;
+
+        var razorDocument = await CreateDocumentAsync(input);
+        var endpoint = new CohostSyntaxTreeEndpoint(IncompatibleProjectService, RemoteServiceInvoker);
+        
+        var request = new SyntaxTreeRequest
+        {
+            TextDocument = new TextDocumentIdentifier { DocumentUri = razorDocument.CreateDocumentUri() }
+        };
+
+        var result = await endpoint.GetTestAccessor().HandleRequestAsync(request, razorDocument, DisposalToken);
+        
+        Assert.NotNull(result);
+        Assert.NotNull(result.Root);
+        Assert.NotEmpty(result.Root.Kind);
+        Assert.True(result.Root.SpanLength > 0);
+        Assert.NotNull(result.Root.Children);
+    }
+
+    private async Task VerifyGeneratedDocumentContentsAsync(string input, GeneratedDocumentKind kind)
+    {
+        var razorDocument = await CreateDocumentAsync(input);
+        var endpoint = new CohostGeneratedDocumentContentsEndpoint(IncompatibleProjectService, RemoteServiceInvoker);
+        
+        var request = new DocumentContentsRequest
+        {
+            TextDocument = new TextDocumentIdentifier { DocumentUri = razorDocument.CreateDocumentUri() },
+            Kind = kind
+        };
+
+        var result = await endpoint.GetTestAccessor().HandleRequestAsync(request, razorDocument, DisposalToken);
+        
+        Assert.NotNull(result);
+        Assert.NotEmpty(result.Contents);
+        Assert.NotEmpty(result.FilePath);
+        
+        // Verify content based on kind
+        switch (kind)
+        {
+            case GeneratedDocumentKind.CSharp:
+                Assert.Contains("message", result.Contents);
```

Comment:
All of the "expect" values here should be passed in to this method, from the actual tests